cellprofiler/modules/measuretexture.py

```diff
@@ -710,201 +539,111 @@ def display(self, workspace, figure):
 
         figure.subplot_table(0, 0, workspace.display_data.statistics, col_labels=workspace.display_data.col_labels)
 
-    def run_one(self, image_name, object_name, scale, angle, workspace):
+    def run_one(self, image_name, object_name, scale, workspace):
         statistics = []
 
         image = workspace.image_set.get_image(image_name, must_be_grayscale=True)
 
         objects = workspace.get_objects(object_name)
+        labels = objects.segmented
 
-        pixel_data = image.pixel_data
-
-        if image.has_mask:
-            mask = image.mask
-        else:
-            mask = None
+        unique_labels = numpy.unique(labels)
+        if unique_labels[0] == 0:
+            unique_labels = unique_labels[1:]
+
+        n_directions = 13 if objects.volumetric else 4
+
+        if len(unique_labels) == 0:
+            for direction in range(n_directions):
+                for feature_name in F_HARALICK:
+                    statistics += self.record_measurement(
+                        image=image_name,
+                        feature=feature_name,
+                        obj=object_name,
+                        result=numpy.zeros((0,)),
+                        scale="{:d}_{:02d}".format(scale, direction),
+                        workspace=workspace
+                    )
 
-        labels = objects.segmented
+            return statistics
 
+        # IMG-961: Ensure image and objects have the same shape.
         try:
-            pixel_data = objects.crop_image_similarly(pixel_data)
+            mask = image.mask if image.has_mask else numpy.ones_like(image.pixel_data, dtype=numpy.bool)
+            pixel_data = objects.crop_image_similarly(image.pixel_data)
         except ValueError:
-            pixel_data, m1 = cellprofiler.object.size_similarly(labels, pixel_data)
+            pixel_data, m1 = cellprofiler.object.size_similarly(labels, image.pixel_data)
 
             if numpy.any(~m1):
-                if mask is None:
-                    mask = m1
+                if image.has_mask:
+                    mask, m2 = cellprofiler.object.size_similarly(labels, image.mask)
+                    mask[~m2] = False
                 else:
-                    mask, m2 = cellprofiler.object.size_similarly(labels, mask)
+                    mask = m1
 
-                    mask[~m2] = False
+        pixel_data[~mask] = 0
+        # mahotas.features.haralick bricks itself when provided a dtype larger than uint8 (version 1.4.3)
```

Comment:
> How much memory is required is unclear. However, I will take the 12-bit range allowance into consideration.

If I’m reading the FAQ correctly, it’s `(ngrey, ngrey)` so 16-bit is `(2^16 − 1, 2^16 − 1)`.